// RouterServer proto

package ServerEngine;

message SCMessage
{
	required int32 iCmd = 1;
	optional bytes strMsgBody = 2;
};

message CSMessage
{
	required int32 iCmd = 1;
	optional bytes strMsgBody = 2;
};

enum enBusyStatus
{
	en_BusyStatus_Free = 0;  // 空闲
	en_BusyStatus_Good = 1;  // 良好
	en_BusyStatus_Hot = 2;   // 火爆
	en_BusyStatus_Full = 3;  // 爆满
	en_BusyStatus_End = 4;
};

// RS协议CMD
enum RouterServerCMD
{
	RS_ROUTER_AUTH = 1000;  // RouterServer的权限认证协议
	RS_ROUTER_WORLDLIST = 1001; // 获取服务器列表
	RS_ROUTER_ALOGIN = 1002; // 打通登录协议
	RS_ROUTER_ERROR = 1003; // 错误协议
};

// RS认证错误码
enum RSAuthCode
{
	RS_AUTU_OK = 0;  // 认证成功
	RS_AUTU_PASSWD = 1; // 密码错误
};

message CS_RS_Auth
{
	required string strAccount = 1;
	required bytes strMd5Passwd = 2;
};

message SC_RS_Auth
{
	required string strAccount = 1;
	required int32 iRetCode = 2;
};

message RSWorld
{
	required int32 iWorldID = 1;
	required string strWorldName = 2;
	required int32 iAtt = 3;
	required int32 iOnlineStatus = 4;
	required int32 iBusyStatus = 5;
};

message SC_RS_WorldList
{
	repeated RSWorld worldList = 1; 
};


message SC_RS_Error
{
	optional string strErrorMsg = 1;
};

message CS_RS_ALogin
{
	required int32 iWorldID = 1;
	required bytes strTransMsg = 2;
};

